sandy/core/cov.py

```diff
@@ -341,6 +343,100 @@ def corr(self):
+        Reduces the size of the matrix, erasing the null values.
+
+        Returns
+        -------
+        nonzero_idxs : `numpy.ndarray`
+            The indices of the diagonal that are not null.
+        cov_reduced : `pandas.DataFrame`
+            The reduced matrix.
+
+        Examples
+        --------
+        >>> S = sandy.CategoryCov(np.diag(np.array([1, 2, 3])))
+        >>> non_zero_index, reduce_matrix = S._reduce_size()
+        >>> non_zero_index
+        array([0, 1, 2], dtype=int64)
+        >>> reduce_matrix
+                      0	          1	          2
+        0	1.00000e+00	0.00000e+00	0.00000e+00
+        1	0.00000e+00	2.00000e+00	0.00000e+00
+        2	0.00000e+00	0.00000e+00	3.00000e+00
+
+        >>> S = sandy.CategoryCov(np.diag(np.array([0, 2, 3])))
+        >>> non_zero_index, reduce_matrix = S._reduce_size()
+        >>> non_zero_index
+        array([1, 2], dtype=int64)
+        >>> reduce_matrix
+                      1	          2
+        1	2.00000e+00	0.00000e+00
+        2	0.00000e+00	3.00000e+00
+        """
+        nonzero_idxs = np.flatnonzero(np.diag(self.data))
+        cov_reduced = self.data.loc[nonzero_idxs, nonzero_idxs]
+        return nonzero_idxs, cov_reduced
+
+    def _restore_size(cls, nonzero_idxs, cov_reduced, dim):
+        """
+        Restore the size of the matrix
+
+        Parameters
+        ----------
+        nonzero_idxs : numpy.ndarray
+            The indices of the diagonal that are not null.
+        cov_reduced : sandy.core.cov._Cov
+            The reduced matrix.
+        dim : int
+            Dimension of the original matrix.
+
+        Returns
+        -------
+        cov : sandy.core.cov._Cov
+            Matrix of specified dimensions.
+
+        """
+        cov = np.zeros((dim, dim))
+        for i, ni in enumerate(nonzero_idxs):
+            cov[ni, nonzero_idxs] = cov_reduced[i]
+        return cov
+
+    def inv(self):
+        """
+        Method for calculating the inverse matrix.
+
+        Returns
+        -------
+        `CategoryCov`
+            The inverse matrix.
+
+        Examples
+        --------
+        >>> S = sandy.CategoryCov(np.diag(np.array([1, 2, 3])))
+        >>> S.inv()
+                    0           1           2
+        0 1.00000e+00 0.00000e+00 0.00000e+00
+        1 0.00000e+00 5.00000e-01 0.00000e+00
+        2 0.00000e+00 0.00000e+00 3.33333e-01
+
+        >>> S = sandy.CategoryCov(np.diag(np.array([0, 2, 3])))
+        >>> S.inv()
+                    0           1           2
+        0 0.00000e+00 0.00000e+00 0.00000e+00
+        1 0.00000e+00 5.00000e-01 0.00000e+00
+        2 0.00000e+00 0.00000e+00 3.33333e-01
+        """
+        index, columns = self.data.index, self.data.columns
+        M_nonzero_idxs, M_reduce = self._reduce_size()
+        unit = np.identity(len(M_reduce))
+        M_reduce_inv = splu(csc_matrix(M_reduce)).solve(unit)
+        data = self._restore_size(M_nonzero_idxs, M_reduce_inv, len(self.data))

```

Comment:
make sure to correct if you change restore size to a classmethod...and also if you change the output type